modules/launcher/src/main/java/org/mule/runtime/module/launcher/service/MuleServiceManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.launcher.service;
+
+import static java.util.Collections.unmodifiableList;
+import static org.mule.runtime.core.util.Preconditions.checkArgument;
+import static org.mule.runtime.module.launcher.MuleFoldersUtil.getServicesFolder;
+import static org.mule.runtime.module.launcher.service.LifecycleFilterServiceProxy.createServiceProxy;
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.core.api.MuleException;
+import org.mule.runtime.core.api.lifecycle.StartException;
+import org.mule.runtime.core.api.lifecycle.Startable;
+import org.mule.runtime.core.api.lifecycle.Stoppable;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service manager to use in the Mule container.
+ */
+public class MuleServiceManager implements ServiceManager
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(MuleServiceManager.class);
+
+    private final ServiceDiscoverer serviceDiscoverer;
+    private List<Service> registeredServices = new ArrayList<>();
+    private List<Service> wrappedServices;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param serviceDiscoverer container service discoverer. Non null.
+     */
+    public MuleServiceManager(ServiceDiscoverer serviceDiscoverer)
+    {
+        checkArgument(serviceDiscoverer != null, "serviceDiscoverer cannot be null");
+        this.serviceDiscoverer = serviceDiscoverer;
+    }
+
+    @Override
+    public void start() throws MuleException
+    {
+        File servicesFolder = getServicesFolder();
+        if (!servicesFolder.exists())
+        {
+            servicesFolder.mkdir();
+        }
+
+        try
+        {
+            registeredServices = serviceDiscoverer.discoverServices();
+            wrappedServices = wrapServices(registeredServices);
+
+            startServices();
```

Comment:
If a Service can use Startable and Stoppable then it should also be able to use Initialisable and Stoppable. If not it's not clear from the user perspective what lifecycle methods can be used. It seems to me that you shouldn't use Stoppable and Startable since those are meant to be used within the Lifecycle context of an artifact. Perhaps it should be better to just support start() and stop() in the Service interface.